server/svix-server/src/v1/endpoints/message.rs

```diff
@@ -263,6 +265,13 @@ async fn create_message(
     // Should never happen since you're giving it an existing Application, but just in case
     .ok_or_else(|| err_generic!("Application doesn't exist: {}", app.id))?;
 
+    if insert_event_type.unwrap_or(false) {
+        if let serde_json::Value::Object(ref mut obj) = data.payload {
```

Comment:
@tasn raised a good point on slack: the payload is enforced to be a JSON object elsewhere in code, but since rust doesn't allow using enum variants as types themselves it must be a `Value`. Question: should I just do an `as_object().unwrap()` here since it's guaranteed to be an object? Or is the current "silently do nothing if this invariant is violated" preferred? Or augment the latter by logging in case the invariant is broken?

Reply:
I actually like what you did. There's no performance difference, but if we ever relaxed the requirements for `data.payload`, it's hard to audit where we made the assumption that it's always an object.

Reply:
Yup, I'm with @svix-gabriel. Though maybe add an error log for the `else` clause? Just so we know if it's ever not the case?

Reply:
Updated with a log message.